src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Compatibility.Common.targets

```diff
@@ -33,6 +34,37 @@
       CompatibilitySuppressionFilePath="$(CompatibilitySuppressionFilePath)"
       BaselinePackageTargetPath="$(PackageValidationBaselinePath)"
       DisablePackageBaselineValidation="$(DisablePackageBaselineValidation)"
-      RoslynAssembliesPath="$(RoslynAssembliesPath)" />
+      RoslynAssembliesPath="$(RoslynAssembliesPath)"
+      ReferenceDirectories="@(PackageValidationReferenceDirectories)" />
+  </Target>
+
+  <PropertyGroup>
+    <_GetPackageValidationReferenceDirectoriesDependsOn Condition="'$(IsCrossTargetingBuild)' != 'true'">_GetReferenceDirectoriesForPackageValidation</_GetPackageValidationReferenceDirectoriesDependsOn>
+    <_GetPackageValidationReferenceDirectoriesDependsOn Condition="'$(IsCrossTargetingBuild)' == 'true'">_ComputeTargetFrameworkItems</_GetPackageValidationReferenceDirectoriesDependsOn>
+  </PropertyGroup>
+
+  <Target Name="_GetReferenceDirectoriesForPackageValidation"
+          DependsOnTargets="ResolveReferences"
+          Returns="@(_ReferenceDirectoriesWithTargetFramework)">
+    <ItemGroup>
+      <_referenceDirectories Include="@(ReferencePath -> '%(RootDir)%(Directory)')" />
+      <_ReferenceDirectoriesWithTargetFramework Include="%(_referenceDirectories.Identity)"  TargetFramework="$(TargetFramework)" />
+    </ItemGroup>
+  </Target>
+
+  <Target Name="_GetPackageValidationReferenceDirectories"
+          DependsOnTargets="$(_GetPackageValidationReferenceDirectoriesDependsOn)"
+          Condition="'$(RunPackageValidationWithoutReferences)' != 'true'">
+
+    <MSBuild Projects="@(_InnerBuildProjects)"
```

Comment:
Can we add a comment about where we copied this technique from (NuGet targets I believe), and what it needs to stay consistent with?

Reply:
Which technique in specific? Is it the fact that we don't build the project references or the DependsOnTargets hooks we picked?